tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -4069,6 +4069,12 @@ class HLSLExternalSource : public ExternalSemaSource {
   {
     DXASSERT_NOMSG(ULE != nullptr);
 
+    // Intrinsics live in the global namespace, so references to their names
+    // should be either unqualified or '::'-prefixed.
+    if (ULE->getQualifier() && ULE->getQualifier()->getKind() != NestedNameSpecifier::Global) {
```

Comment:
Does fxc allow overloading intrinsic from non-global namespace? Also, what happens if we try to overload an intrinsic from the global namespace? I think fxc allows that? If yes, should we have an xfailing test to cover that scenario and address it later?

Reply:
Added more tests to cover those cases, and filed #1887